torch/utils/checkpoint.py

```diff
@@ -1,15 +1,138 @@
 import torch
 import warnings
 import weakref
-from typing import Any, Callable, ContextManager, Iterable, List, Tuple
+from typing import Any, Callable, ContextManager, Iterable, List, Sequence, Tuple, Union
 import contextlib
+from torch.utils._python_dispatch import TorchDispatchMode
+from torch.utils._pytree import tree_map
 
 __all__ = [
     "checkpoint", "checkpoint_sequential", "CheckpointFunction",
     "check_backward_validity", "detach_variable", "get_device_states",
-    "set_device_states",
+    "set_device_states", "list_operators", "get_selective_checkpoint_context_fn"
 ]
 
+def list_operators(function, *args, **kwargs):
+    """
+    Utility to return the list of operators invoked by :attr:`function` that can
+    be used with :func:`get_selective_checkpoint_context_fn`
+
+    Example::
+
+        >>> list_operators(lambda x: x.clone().sin(), torch.tensor(1.))
+        [<OpOverload(op='aten.clone', overload='default')>, <OpOverload(op='aten.sin', overload='default')>]
```

Comment:
so if I want to selectively checkpoint the clone, I'd want to do `torch.ops.aten.clone.default` as my operator arg?

Reply:
correct